src/Dialogs/Message.cpp

```diff
@@ -156,74 +156,53 @@ ShowMessageBox(const TCHAR *text, const TCHAR *caption, unsigned flags)
 
 
 
-int    //////// LOCKSCREEN
+void
 ShowLockBox()
 {
-
   SingleWindow &main_window = UIGlobals::GetMainWindow();
 
   const unsigned button_width = Layout::Scale(15u);
-  const unsigned button_height = Layout::Scale(15u);
+  const unsigned button_height = Layout::GetMinimumControlHeight();
 
   // Create dialog
   WindowStyle style;
-  style.Hide();
   style.ControlParent();
 
   const DialogLook &dialog_look = UIGlobals::GetDialogLook();
 
+  const PixelSize root_size = main_window.GetSize();
+  
+  const int dialog_x = (root_size.cx - button_width) / 4;
+  const int dialog_y = (root_size.cy - button_height) / 4 * 3;
+
   PixelRect form_rc;
-  form_rc.left = 0;
-  form_rc.top = 0;
-  form_rc.right = button_width;
-  form_rc.bottom = button_height;
+  form_rc.left = dialog_x;
+  form_rc.top = dialog_y;
+  form_rc.right = dialog_x + button_width;
+  form_rc.bottom = dialog_y + button_height;
 
   WndForm wf(main_window, dialog_look, form_rc, "", style);
 
   ContainerWindow &client_area = wf.GetClientAreaWindow();
   
-  
-//  const int dialog_x = Layout::Scale(44u); 
-//  const int dialog_y = Layout::Scale(44u); 
-  const PixelSize root_size = main_window.GetSize();
-  
-  const int dialog_x = (root_size.cx - button_width) / 4;
-  const int dialog_y = (root_size.cy - button_height) / 4 * 3;
-  wf.Move(dialog_x, dialog_y, button_width, button_height);
-
   PixelRect button_rc;
   button_rc.left = 0;
   button_rc.top = 0; 
-  button_rc.right = button_rc.left + button_width;
-  button_rc.bottom = button_rc.top + button_height;
+  button_rc.right = button_width;
+  button_rc.bottom = button_height;
 
-  // Create buttons
+  // Create button
   WindowStyle button_style;
   button_style.TabStop();
+  
+  const TCHAR * caption = "U";
+  const Button * button = new Button(client_area, dialog_look.button, caption, button_rc, button_style, wf, IDOK);
```

Comment:
Don't allocate the `Button` dynamically. It can perfectly live on the stack, e.g. simply like this:
`const Button button(client_area, dialog_look.button, caption, button_rc, button_style, wf, IDOK);`
Now you don't even need to `delete` it, because the C++ compiler will automatically do that for you.